lib/services/internalEngine/index.js

```diff
@@ -79,10 +79,21 @@ class InternalEngine {
     this.bootstrap = bootstrap;
 
     if (!this.client) {
-      this.client = new es.Client({
-        hosts: this.config.hosts || this.config.host + ':' + this.config.port,
-        apiVersion: this.config.apiVersion
-      });
+      let options;
+
+      if (this.config.client) {
+        options = Object.assign({}, this.config.client);
+      }
+      else {
+        // keep compatibility layer for the time being
+        // @todo: once the install base is clean, remove this
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)